docs/deployment/kubernetes-deployment.md

```diff
@@ -0,0 +1,384 @@
+
+### Option 1: Deploy using Aspir8
+
+[Aspir8](https://prom3theu5.github.io/aspirational-manifests/) is an open-source project that automatically generates Kubernetes deployment YAML based on your .NET Aspire app host manifest. This is the most straightforward approach for getting started.
+
+#### Install Aspir8
+
+Install the Aspir8 global tool:
+
+```dotnetcli
+dotnet tool install -g aspirate
+```
+
+#### Generate and deploy manifests
+
+1. Navigate to your .NET Aspire project's app host directory:
+
+   ```bash
+   cd YourAspireApp.AppHost
+   ```
+
+1. Initialize Aspir8 in your project:
+
+   ```dotnetcli
+   aspirate init
+   ```
+
+   This creates an `aspirate.json` configuration file where you can customize deployment settings.
+
+1. Generate Kubernetes manifests:
+
+   ```dotnetcli
+   aspirate generate
+   ```
+
+   This command:
+   - Builds your .NET Aspire projects
+   - Creates container images  
+   - Generates Kubernetes YAML manifests in the `manifests` folder
+
+1. Review the generated manifests in the `manifests` folder. You'll find:
+   - `Deployment` objects for your application services
+   - `Service` objects for networking
+   - `ConfigMap` objects for configuration
+   - Container registry configurations
+
+1. Apply the manifests to your Kubernetes cluster:
+
+   ```dotnetcli
+   aspirate apply
+   ```
+
+#### Manage deployments
+
+Aspir8 provides additional commands for managing your deployments:
+
+- **Update deployment**: Run `aspirate generate` and `aspirate apply` again after making code changes
+- **Remove deployment**: Use `aspirate destroy` to clean up all resources
+
+### Option 2: Use .NET Aspire 9.3+ built-in Kubernetes publishing
+
+.NET Aspire 9.3 introduced native Kubernetes manifest generation capabilities that allow you to customize the output programmatically.
+
+#### Configure Kubernetes publishing
+
+In your app host project, you can configure Kubernetes-specific settings:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Configure Kubernetes environment
+builder.AddKubernetesEnvironment("prod")
+       .WithProperties(env =>
+       {
+           env.DefaultImagePullPolicy = "Always";
+       });
+
+// Add your services with Kubernetes-specific customizations
+var apiService = builder.AddProject<Projects.YourApp_ApiService>("apiservice")
+    .PublishAsKubernetesService(resource =>
+    {
+        // Customize deployment settings
+        resource.Deployment!.Spec.RevisionHistoryLimit = 5;
+        
+        // Add labels or annotations
+        resource.Service!.Metadata.Labels.Add("app", "your-api");
+    });
+
+var webService = builder.AddProject<Projects.YourApp_Web>("webfrontend")
+    .WithReference(apiService)
+    .WithExternalHttpEndpoints()
+    .PublishAsKubernetesService(resource =>
+    {
+        // Configure service type for external access
+        resource.Service!.Spec.Type = "LoadBalancer";
+    });
+
+builder.Build().Run();
+```
+
```

Comment:
Please describe what this code is doing, and when you mention an API, please do so using the `<xref:...>` syntax.